### DISCUSSION TITLE
Craft net: Make auto-renew actually renew plugins first

### DISCUSSION THREAD

**Discussion by darylknight** [craftcms/cms#11651] [Ideas]
Sorry this is on the wrong repository. I thought there used to be a "Craft net" repo on here but I can't find it anymore.

A client's plugin licenses recently expired (they have a "forever" Craft CMS 2 license), and I instructed them to go to id.craftcms.com to renew them. It looks like what they did was just turn on the "auto renew" light switch and expected that to fix the problem.

Whilst I understand why that didn't do anything (the switch just enables auto renew "next time"), it would make sense to change this functionality so that if a plugin is expired, and a client enabled auto renew, that it renews them first and charges them, _then_ enables auto-renew? As a result, I needed to email them again and explain that their plugins were still expired, they're just now set to auto-renew after the next time they've manually renewed them, which seems a bit backwards?

**Comment by brandonkelly** [comment]
Hm… I’m not sure that would be much of an improvement, since you’re not actually saying you want to purchase anything immediately when enabling auto-renew. Maybe we should just hide that setting altogether for expired licenses, and give users the ability to opt into auto-renew whenever they are manually renewing.

> Sorry this is on the wrong repository. I thought there used to be a "Craft net" repo on here but I can't find it anymore.

It’s on the @pixelandtonic org: https://github.com/pixelandtonic/craftnet

**Comment by darylknight** [comment]
Yeah, I think that would be a great solution, thank you for considering it. It makes sense to me that simply enabling a toggle switch on an expired plugin wouldn't do anything - I'm just trying to over-engineer for people who aren't paying attention. I guess we can only go so far!
Removing that option for expired plugins, and asking them to enable auto-renew as part of the renewal process would be awesome.
